Specify before hook context merging behavior (#72)
open-feature/spec: specification/flag-evaluation/hooks.md

##### Requirement 3.2

> The `before` stage **MUST** run before flag evaluation occurs. It accepts a `hook context` (required) and `HookHints` (optional) as parameters and returns either an `EvaluationContext` or nothing.

```
EvaluationContext|void before(HookContext, HookHints)
```

##### Requirement 3.3

> Any `EvaluationContext` returned from a `before` hook **MUST** be passed to subsequent `before` hooks (via `HookContext`).

##### Requirement 3.4

> > When `before` hooks have finished executing, any resulting `EvaluationContext` **MUST** be merged with the invocation `EvaluationContext` wherein the invocation `EvaluationContext` win any conflicts.

Reviewer: ```suggestion
> When `before` hooks have finished executing, any resulting `EvaluationContext` **MUST** be merged with the invocation `EvaluationContext` with the invocation `EvaluationContext` taking precedence in the case of any conflicts.
```
